### PR TITLE
Automation/Script editor add settings,category overflow actions

### PR DESCRIPTION
**Pull request by wendevlin**
## Proposed change
- Add settings action in automation & script editor
- Add category assign/edit action in automation & script editor

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/automation/ha-automation-editor.ts [hunk 1]
```diff
@@ -492,6 +525,16 @@ export class HaAutomationEditor extends PreventUnsavedMixin(
     }
   }
 
+  public hassSubscribe() {
+    return [
+      subscribeEntityRegistry(this.hass.connection, (entries) => {
```
Comment (bramkragten) [inline src/panels/config/automation/ha-automation-editor.ts:530]: Dont use subscribe, use the full entities context that is available for all config pages:

```ts
  @state()
  @consume({ context: fullEntitiesContext, subscribe: true })
  _entityReg!: EntityRegistryEntry[];
  ```

Comment (wendevlin) [reply src/panels/config/automation/ha-automation-editor.ts:530]: cool thanks, this should be also changed for scenes then?

Comment (bramkragten) [reply src/panels/config/automation/ha-automation-editor.ts:530]: And if you want just optimize is, and only update on the changes of 1 entry, do:

```ts
  @consume({ context: fullEntitiesContext, subscribe: true })
  @transform<EntityRegistryEntry[], EntityRegistryEntry>({
    transformer: (this, value)  =>
      value.find(
          ({ entity_id }) => entity_id === this._entityId
        );
  })

File: src/panels/config/automation/ha-automation-editor.ts [hunk 2]
```diff
@@ -172,6 +183,28 @@ export class HaAutomationEditor extends PreventUnsavedMixin(
             ></ha-svg-icon>
           </ha-list-item>
 
+          <ha-list-item
+            graphic="icon"
+            .disabled=${!stateObj}
```
Comment (bramkragten) [inline src/panels/config/automation/ha-automation-editor.ts:188]:
```suggestion
            .disabled=${!this._registryEntry}
```

File: src/panels/config/automation/ha-automation-editor.ts [hunk 3]
```diff
@@ -172,6 +183,28 @@ export class HaAutomationEditor extends PreventUnsavedMixin(
...
+          >
+            ${this.hass.localize(
+              "ui.panel.config.automation.picker.show_settings"
+            )}
+            <ha-svg-icon slot="graphic" .path=${mdiCog}></ha-svg-icon>
+          </ha-list-item>
+
+          <ha-list-item
+            graphic="icon"
+            .disabled=${!stateObj}
```
Comment (bramkragten) [inline src/panels/config/automation/ha-automation-editor.ts:199]:
```suggestion
            .disabled=${!this._registryEntry}
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**